src/CloudentityAuth.js

```diff
@@ -92,7 +95,15 @@ class CloudentityAuth {
    */
   authorize (dynamicOptions) {
     const dynamicScopes = dynamicOptions && dynamicOptions.scopes && notEmptyStringArray(dynamicOptions.scopes);
-    const finalOptions = dynamicScopes ? {...this.options, ...{scopes: dynamicOptions.scopes}} : this.options;
+    const dynamicResponseType = dynamicOptions && dynamicOptions.responseType && notEmptyStringArray(dynamicOptions.responseType);
+
+    const finalOptions = Object.assign(
+      {},
+      this.options,
+      dynamicScopes ? {scopes: dynamicOptions.scopes} : {},
+      dynamicResponseType ? {responseType: dynamicOptions.responseType} : {},
+    )
```

Comment:
you can use object spreading - const finalOptions = { ...this.options, ...(dynamicScopes ? {scopes: dynamicOptions.scopes} : {}), ...(dynamicResponseType ? {responseType: dynamicOptions.responseType} : {}) }